Address Safer CPP warnings in VTTRegion

eric-carlson opened the pull request:
#### a211c58c1873e2a9dbb6a133c8b19608ffd3c6cf
<pre>
Address Safer CPP warnings in VTTRegion
<a href="https://bugs.webkit.org/show_bug.cgi?id=289373">https://bugs.webkit.org/show_bug.cgi?id=289373</a>
<a href="https://rdar.apple.com/146516346">rdar://146516346</a>

Reviewed by Chris Dumez.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::protectedClassList):
* Source/WebCore/dom/Element.h:
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::displayLastTextTrackCueBox):
(WebCore::VTTRegion::willRemoveTextTrackCueBox):
(WebCore::VTTRegion::getDisplayTree):
(WebCore::VTTRegion::prepareRegionDisplayTree):
(WebCore::VTTRegion::protectedDocument const):
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/html/track/VTTRegion.h:

Canonical link: <a href="https://commits.webkit.org/292029@main">https://commits.webkit.org/292029@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/89a648cc998c4d61c200593384fa54fd2fb2f37c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94738 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14329 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4161 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99757 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45230 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96788 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14605 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22748 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/72276 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29580 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97740 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10880 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85538 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52607 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10572 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3248 "Failed to checkout and rebase branch from PR 42114") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44568 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80794 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3353 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101799 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21768 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15890 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81278 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22016 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81563 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/80656 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20139 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25203 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/2611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15001 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21745 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26858 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21406 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24877 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23145 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/track/VTTRegion.cpp:
```diff
@@ -252,6 +252,7 @@ void VTTRegion::appendTextTrackCueBox(Ref<TextTrackCueBox>&& displayBox)
 {
     ASSERT(m_cueContainer);
 
+    Ref protectedThis { *this };
```
All these protectors are wrong. Per Safer CPP guidelines, we now protect the objects at call sites. Protectors are no longer a thing.

cdumez commented on Source/WebCore/html/track/VTTRegion.cpp:
```diff
@@ -252,10 +252,11 @@ void VTTRegion::appendTextTrackCueBox(Ref<TextTrackCueBox>&& displayBox)
     ASSERT(m_cueContainer);
 
-    if (m_cueContainer->contains(displayBox.ptr()))
+    RefPtr cueContainer = m_cueContainer;
```
You can just mark `m_cueContainer` as `const` and then use:
```
lazyInitialize(m_cueContainer, HTMLDivElement::create(downcast<Document>(*scriptExecutionContext())));
```

Since `m_cueContainer` never gets updated, just lazily initialized.

cdumez commented on Source/WebCore/html/track/VTTRegion.cpp:
```diff
@@ -264,17 +265,19 @@ void VTTRegion::displayLastTextTrackCueBox()
 
     // The container needs to be rendered, if it is not empty and the region is not currently scrolling.
-    if (!m_cueContainer->renderer() || !m_cueContainer->hasChildNodes() || m_scrollTimer.isActive())
+    RefPtr cueContainer = m_cueContainer;
```
Should no longer be needed with my suggestion above.

cdumez commented on Source/WebCore/html/track/VTTRegion.cpp:
```diff
@@ -264,17 +265,19 @@ void VTTRegion::displayLastTextTrackCueBox()
+        cueContainer->classList().add(textTrackCueContainerScrollingClass());
 
-    float regionBottom = m_regionDisplayTree->boundingClientRect().maxY();
+    RefPtr regionDisplayTree = m_regionDisplayTree;
```
`m_regionDisplayTree` can be marked as `const` and lazily initialized too.

cdumez reviewed (approved):
LGTM

cdumez commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -4830,6 +4830,11 @@ DOMTokenList& Element::classList()
 
+Ref<DOMTokenList> Element::protectedClassList()
+{
+    return Ref { classList() };
```
`Ref { }` is not needed.

cdumez reviewed (approved):
Still LGTM

eric-carlson committed:
Address Safer CPP warnings in VTTRegion
https://bugs.webkit.org/show_bug.cgi?id=289373
rdar://146516346

Reviewed by Chris Dumez.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::protectedClassList):
* Source/WebCore/dom/Element.h:
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::displayLastTextTrackCueBox):
(WebCore::VTTRegion::willRemoveTextTrackCueBox):
(WebCore::VTTRegion::getDisplayTree):
(WebCore::VTTRegion::prepareRegionDisplayTree):
(WebCore::VTTRegion::protectedDocument const):
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/html/track/VTTRegion.h:

Canonical link: https://commits.webkit.org/292029@main
